### PR TITLE
WIP: bump k8s to 1.8.0-beta.1

### PR DESCRIPTION
**Pull request by rphillips**
Issues:

- [ ] modify cluster/manifests/kube-system-rbac-role-binding.yaml to use `rbac.authorization.k8s.io/v1beta1` instead of v1alpha1
  - v1beta1 changes the selectors to be immutable

### CODE REVIEW

**Comment by squat** [comment]
Awesome

**Comment by klausenbusk** [comment]
The `apiVersion` of DaemonSet/Deployment should also be updated.

https://github.com/kubernetes/features/blob/master/release-1.8/release_notes_draft.md#major-themes

File: pkg/asset/internal/templates.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ contexts:
     user: kubelet
 `)
 
-var KubeSystemSARoleBindingTemplate = []byte(`apiVersion: rbac.authorization.k8s.io/v1alpha1
+var KubeSystemSARoleBindingTemplate = []byte(`apiVersion: rbac.authorization.k8s.io/v1beta1
```
Comment (klausenbusk) [inline pkg/asset/internal/templates.go]: Should be `v1` per the release notes draft: https://github.com/kubernetes/features/blob/master/release-1.8/release_notes_draft.md#auth-1

**Comment by ericchiang** [comment]
Seeing this error when trying to use this PR

```
Ignoring json-2.0.3 because its extensions are not built.  Try: gem pristine json --version 2.0.3
Ignoring nokogiri-1.6.8.1 because its extensions are not built.  Try: gem pristine nokogiri --version 1.6.8.1
Starting temporary bootstrap control plane...
Waiting for api-server...
Creating self-hosted assets...
	created          kube-apiserver secret
	created kube-controller-manager poddisruptionbudget
	created kube-controller-manager secret
	created kube-controller-manager deployment
	created                kube-dns deployment
	created                kube-dns service
	created        kube-flannel-cfg configmap
	created          kube-scheduler poddisruptionbudget
	created          kube-scheduler deployment
	created       system:default-sa clusterrolebinding
Error creating assets: [error when creating "/home/core/cluster/manifests/kube-apiserver.yaml": DaemonSet.apps "kube-apiserver" is invalid: spec.template.metadata.labels: Invalid value: map[string]string{"k8s-app":"kube-apiserver", "tier":"control-plane"}: `selector` does not match template `labels`, error when creating "/home/core/cluster/manifests/kube-flannel.yaml": DaemonSet.apps "kube-flannel" is invalid: spec.template.metadata.labels: Invalid value: map[string]string{"k8s-app":"flannel", "tier":"node"}: `selector` does not match template `labels`, error when creating "/home/core/cluster/manifests/kube-proxy.yaml": DaemonSet.apps "kube-proxy" is invalid: spec.template.metadata.labels: Invalid value: map[string]string{"k8s-app":"kube-proxy", "tier":"node"}: `selector` does not match template `labels`, error when creating "/home/core/cluster/manifests/pod-checkpointer.yaml": DaemonSet.apps "pod-checkpointer" is invalid: spec.template.metadata.labels: Invalid value: map[string]string{"tier":"control-plane", "k8s-app":"pod-checkpointer"}: `selector` does not match template `labels`]

NOTE: Bootkube failed to create some cluster assets. It is important that manifest errors are resolved and resubmitted to the apiserver.
For example, after resolving issues: kubectl create -f <failed-manifest>

	Pod Status: kube-controller-manager	Pending
	Pod Status:        pod-checkpointer	DoesNotExist
	Pod Status:          kube-apiserver	DoesNotExist
	Pod Status:          kube-scheduler	Pending
```

File: pkg/asset/internal/templates.go (hunk 1 above)
Comment (ericchiang) [reply pkg/asset/internal/templates.go]: Either works. v1alpha1 doesn't anymore due to https://github.com/kubernetes/kubernetes/pull/51839

**Comment by rphillips** [comment]
Reproduced... It has something to do with defaulting labels now. Looking into it.

[Reference](https://github.com/kubernetes/kubernetes/pull/50719)

**Comment by rphillips** [comment]
@ericchiang fixed... updated the selectors on the daemonsets.

File: pkg/asset/internal/templates.go [hunk 2]
```diff
@@ -1193,7 +1205,7 @@ data:
     }
 `)
 
-var KubeCalicoTemplate = []byte(`apiVersion: extensions/v1beta1
```
Comment (dghubble) [inline pkg/asset/internal/templates.go]: Look for CalicoNodeTemplate, this was renamed on master

**Comment by ericchiang** [comment]
@rphillips can we update this PR now that 1.8 is out? (cc @squat)

**Comment by squat** [comment]
1.8.0 hyperkube is about to get cut. Waiting for the necessary LGTM and it will be available. xref https://github.com/coreos/kubernetes/pull/167

**Comment by rphillips** [comment]
yep.. then I'll tweak this PR to remove the API bumps, and do a migration of the APIs in a separate PR

**Comment by rphillips** [comment]
closing in favor of https://github.com/kubernetes-incubator/bootkube/pull/729

**Pull request closed without merging**